src/runner.ts

```diff
@@ -1,118 +1,144 @@
+
+        logger.verbose("Origin: %o, Destination: %o, SearchDate: %o, StartDate: %o, EndDate: %o, NotVias: %o",
+            orig, dest, searchDate, startDate, endDate, notVias);
+
+        const journeys = query.plan(orig, dest, searchDate, startDate, endDate, notVias);
+        res.send(journeys);
+    });
+
+    app.get("/detail", (req, res, next) => {
+
+        logger.verbose("Request to '/detail' with req.query: %s", JSON.stringify(req.query));
+        const orig = req.query.orig;
+        const dest = req.query.dest;
+        const startDate = new Date(req.query.startDate);
+        const endDate = new Date(req.query.endDate);
+        const searchDate = getMidnight(startDate.toISOString());
+
+        const notVias = req.query.notVia ? req.query.notVia : [];
+        if (!validateNotVias(notVias, orig, dest)) {
+            sendInvalidNotViasMessage(res, notVias, orig, dest);
+            return;
+        }
+
+        logger.verbose("Origin: %o, Destination: %o, SearchDate: %o, StartDate: %o, EndDate: %o, NotVias: %o",
+            orig, dest, searchDate, startDate, endDate, notVias);
+
+        const journeys = detailedQuery.plan(orig, dest, searchDate, startDate, endDate, notVias);
+        res.send(journeys);
+    });
+
+    app.get("/first-arrival", (req, res, next) => {
+
+        logger.verbose("Request to '/first-arrival' with req.query: %s", JSON.stringify(req.query));
+        const orig = req.query.orig;
+        const dest = req.query.dest;
+        const startDate = new Date(req.query.startDate);
+
+        const notVias = req.query.notVia ? req.query.notVia : [];
+        if (!validateNotVias(notVias, orig, dest)) {
+            sendInvalidNotViasMessage(res, notVias, orig, dest);
+            return;
+        }
+        const searchDate = getMidnight(startDate.toISOString());
+
+        logger.verbose("Single Lookup / Origin: %o, Destination: %o, SearchDate: %o, StartDate: %o, NotVias: %o",
+            orig, dest, searchDate, startDate, notVias);
+
+        const midnight = getMidnight(startDate.toISOString());
+        const startSeconds = (startDate.valueOf() - midnight.valueOf()) / 1000;
+        const journeys = singleLookup.plan(orig, dest, searchDate, startSeconds, notVias);
+        res.send(journeys);
+    });
+
+    app.get("/get-log-level", (req, res, next) => {
+        logger.verbose("Request to '/get-log-level' with req.query: %s", JSON.stringify(req.query));
+        res.send(logger.transports[0].level);
+    });
+
+    app.get("/set-log-level", (req, res, next) => {
+        logger.verbose("Request to '/set-log-level' with req.query: %s", JSON.stringify(req.query));
+
+        const level = req.query.level;
+        if (!level || !logLevels.includes(level)) {
+            res.status(500).send("incorrect/missing query param [level], possible values are: [ " + logLevels + " ]");
+        } else {
+            const prevLevel = logger.transports[0].level;
+            logger.transports[0].level = level;
```

Comment:
We should be able to support multiple transports, especially in the scenario of having a console logger as well as file logger.
Maybe a method in the `logger.ts` file would work.

Reply:
Is it sufficient to set the same log level for all transports, or do you mean being able to individually configure each?

Reply:
I think changing the log level for all of the loggers is fine.

Reply:
1caac87
--


 Added getters/setters to change the log level for all Logger Transports.